lib/Xmds/Soap4.php

```diff
@@ -272,6 +273,16 @@ public function RegisterDisplay($serverKey, $hardwareKey, $displayName, $clientT
         $display->clientVersion = $clientVersion;
         $display->clientCode = $clientCode;
         //$display->operatingSystem = $operatingSystem;
+
+        // Parse operatingSystem data
+        $operatingSystem = json_decode($operatingSystem, true);
```

Comment:
Newer versions of players will submit a JSON string to this endpoint, and in that case this code looks good. However older players do not know how to do that yet, and so the JSON decode may fail.

We should handle that case, and instead take the original $operatingSystem as provided by the player, and submit it to osVersion.

Reply:
Thank you again for the review! Feedback has been addressed [here](https://github.com/xibosignage/xibo-cms/pull/2574/commits/d783e966a3136dce6a1006af8fdca48b4cfe7c0f). Requesting re-review, please. Thanks!